### DISCUSSION TITLE
[ML] Spelling Auto Correction

### DISCUSSION THREAD

**Discussion by FaizFaisalAhmad** [teamalgoritma/community#669] [Q&A] [answered]
Apakah ada referensi yang dapat saya baca terkait model machine learning untuk spelling auto correction?

**Comment by nadyadtm** [comment]
Dear Pak @FaizFaisalAhmad 
Baik pak, kami akan mencarikan terlebih dahulu untuk referensinya

**Comment by nadyadtm** [comment]
Dear Pak @FaizFaisalAhmad 

Untuk membuat autocorrection dapat menggunakan perhitungan similarity, yaitu melihat kemiripan suatu kata dengan kata yang lain. Untuk mengetahui langkah-langkahnya dapat merujuk pada link berikut ini

https://predictivehacks.com/how-to-build-an-autocorrect-in-python/

Semoga membantu

**Reply by FaizFaisalAhmad** [reply]
baik saya coba pelajari dahulu, terima kasih mbak nadya

**Comment by vidric** [comment, marked as answer]
tergantung sumber datanya Pak. Kalau mengambil dari artikel berita, bisa pakai sastrawi. Tapi kalau sumber datanya hasil ketikan manusia seperti review, twitter, saran saya manual. Lakukan tokenisasi, lihat frekuensi setiap kata. Lalu bikin dataset untuk converternya dengan mengutamakan kata dengan frekuensi tinggi.

**Reply by FaizFaisalAhmad** [reply]
ini ketikan manual sih, jd ada dokumen yang dibuat oleh manusia sebelum di submit, sistem bisa mendeteksi kata2 yang kurang tepat. dan memberikan beberapa perbaikan yang bisa digunakan.
